odoo/models.py

```diff
@@ -2413,6 +2418,26 @@ def _read_group_format_result(self, data, annotated_groupbys, groupby, domain):
         del data['id']
         return data
 
+    @api.model
+    def _read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        """
+        Executes exactly what the public read_group, except it doesn't order the manyToOne on their names but on their ID instead
+        """
+        if not orderby:
+            if isinstance(groupby, str):
+                groupby = [groupby]
+            groupby_list = groupby[:1] if lazy else groupby
+            order_list = []
+            for order_spec in groupby_list:
+                [field_name, agg] = order_spec.split(":") if ":" in order_spec else [order_spec, ""]
+                if self._fields[field_name].type == 'many2one':
+                    new_order_spec = field_name+".id" + (":" if agg else "") + agg
+                else:
+                    new_order_spec = order_spec
+                order_list.append(new_order_spec)
```

Comment:
Looks overengineered to me, since projections are only available for date and datetime fields.
```suggestion
                field_name = order_spec.split(":")[0]
                if self._fields[field_name].type == 'many2one':
                    order_spec = f"{field_name}.id"  # do not order by comodel's order
                order_list.append(order_spec)
```